app/controllers/inventories_controller.rb

```diff
@@ -1,37 +1,29 @@
 class InventoriesController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_model, only: %i[show destroy restore]
+  before_action :set_model, only: %i[destroy restore]
+  before_action :set_data, only: %i[new create]
 
   # GET /inventories
   # GET /inventories.json
   def index
-    @warehouse = Phone.joins(:inventories, :memory, :model).select(:model_id, :name, 'SUM(quantity) as total_quantity', 'MAX(price) as max_price', 'MIN(price) as min_price', 'MAX(amount) as max_amount', 'MIN(amount) as min_amount').group(:model_id) # rubocop:disable Layout/LineLength
+    @warehouse = Phone.joins(:memory, :inventories, :model).select(:model_id, :name, 'SUM(quantity) as total_quantity', 'MAX(price) as max_price', 'MIN(price) as min_price', 'MAX(amount) as max_amount', 'MIN(amount) as min_amount').where('inventories.discarded_at IS NULL').where('models.discarded_at IS NULL').group(:model_id) # rubocop:disable Layout/LineLength
   end
 
-  # GET /inventories/1
-  # GET /inventories/1.json
-  def show; end
-
   # GET /inventories/new
-  def new
-    @inventory = Inventory.new
-    @models = Model.joins(:manufacturer, :os_name).pluck('models.name', 'manufacturers.name', 'os_names.name').collect do |item|
-      "#{item[0]} (#{item[1]} - #{item[2]})"
-    end.to_json.html_safe
-    @body_colors = BodyColor.pluck(:name).to_json.html_safe
-    @memories = Memory.pluck(:display_name).to_json.html_safe
-  end
+  def new; end
 
   # POST /inventories
   # POST /inventories.json
   def create
     respond_to do |format|
-      if CreateBulkService.call(convert_params, current_user)
+      begin
+        Inventories::BulkCreateService.call(convert_params, current_user)
         format.html { redirect_to inventories_path, notice: 'Inventory was successfully imported.' }
         format.json { render :index, status: :created, location: inventories_path }
-      else
+      rescue ActiveRecord::RecordInvalid => e
+        flash.now[:alert] = e.message
         format.html { render :new }
-        format.json { render json: 'error', status: :unprocessable_entity }
+        format.json { render json: @inventory, status: :unprocessable_entity }
```

Comment:
It looks weird to return `@inventory` object in spite process is failing 🤔 
Usually when error happen, only error message should be returned.